### PR TITLE
Add subnet and creator restrictions for start, stop, extend

### PR DESCRIPTION
**Pull request by marcesher**

### CODE REVIEW

File: scripts/ec2/stop.coffee [hunk 1]
```diff
@@ -2,58 +2,62 @@
...
+
 getArgParams = (arg) ->
   dry_run = if arg.match(/--dry-run/) then true else false
+  return {dry_run: dry_run}
+
+stopInstances = (msg, params, instances, err) ->
+  return (err) ->
+    if err
+      msg.send "Error! #{err}"
+      return
```
Comment (contolini) [inline scripts/ec2/stop.coffee]: Doesn't really matter but these one-liner `if` statements can be condensed into:

```
return msg.send "Error! #{err}" if err
```

File: scripts/ec2/start_instance.coffee [hunk 2]
```diff
@@ -8,13 +8,38 @@
...
+
+    if dry_run
+      msg.send util.inspect(params, false, null)
+
+    ec2.startInstances params, (err, res) ->
+      if err
+        msg.send "Error: #{err}"
+      else
+       msg.send "Success! The instances are starting"
+       msg.send util.inspect(res, false, null)
```
Comment (contolini) [inline scripts/ec2/start_instance.coffee]: In the same vein it's a bit more idiomatic to use the return-early pattern when possible:

```
return msg.send "Error: #{err}" if err

msg.send "Success! The instances are starting"
msg.send util.inspect(res, false, null)
```

File: INSTALL.md [hunk 3]
```diff
@@ -1,3 +1,66 @@
...
+```
+export HUBOT_AWS_REGION="us-east-1"
+export HUBOT_AWS_EC2_RUN_CONFIG="ec2.json"
+export HUBOT_AWS_CAN_ACCESS_ROLE="ec2"
+export HUBOT_AUTH_ADMIN="1"
+export HUBOT_AWS_DEFAULT_CREATOR_EMAIL="ask-for-it"
+export HUBOT_AWS_ACCESS_KEY_ID="ask-for-it"
+export HUBOT_AWS_SECRET_ACCESS_KEY="ask-for-it"
+
+export HUBOT_AWS_EC2_RUN_USERDATA_PATH=ec2-user-data.txt
```
Comment (contolini) [inline INSTALL.md:51]: Is it worth mentioning the new [HUBOT_AWS_EC2_REQUIRE_SSH_KEY](https://github.com/marcesher/hubot-aws/blob/subnet_restriction/scripts/ec2/run.coffee#L8) variable?

Comment (marcesher) [inline INSTALL.md:51]: I don't know anything about that variable. But whoever added that variable should certainly mention it

File: scripts/ec2/start_instance.coffee (hunk 2 above)
Comment (marcesher) [inline scripts/ec2/start_instance.coffee]: @contolini done in all instances. Thanks for the tip.

**Comment by contolini** [comment]
👍  I don't have AWS console access to end-to-end test this but the code looks good!

**Pull request merged**